### PR TITLE
Close ResponseBody in sample

### PR DESCRIPTION
**Pull request by NightlyNexus**

### CODE REVIEW

File: samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java [hunk 1]
```diff
@@ -26,16 +27,19 @@ public static void main(String... args) throws Exception {
     OkHttpClient client = new OkHttpClient();
 
     // Create request for remote resource.
-    Request request = new Request.Builder()
-        .url(ENDPOINT)
+    Request request = new Request.Builder() //
```
Comment (swankjesse) [inline samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java]: please no `//`

File: samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java [hunk 2]
```diff
@@ -26,16 +27,19 @@ public static void main(String... args) throws Exception {
...
     // Execute the request and retrieve the response.
     Response response = client.newCall(request).execute();
 
     // Deserialize HTTP response to concrete type.
-    Reader body = response.body().charStream();
-    List<Contributor> contributors = GSON.fromJson(body, CONTRIBUTORS.getType());
+    ResponseBody body = response.body();
+    Reader charStream = body.charStream();
+    List<Contributor> contributors = GSON.fromJson(charStream, CONTRIBUTORS.getType());
+    charStream.close();
```
Comment (swankjesse) [inline samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java]: you don't need both of these `close()` calls

Comment (swankjesse) [inline samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java]: probably `body.close()` is enough

Comment (NightlyNexus) [inline samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java]: I know I sound dumb, but I don't really understand.
Can you ELI5 for my benefit?

Comment (swankjesse) [inline samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java]: There’s no need to call `close()` on the charStream. Closing the body does that for you.

Comment (NightlyNexus) [inline samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java]: Right, but I don't understand how.
ResponseBody closes the BufferedSource, but how does it close the Reader for the char stream?

Comment (JakeWharton) [inline samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java]: The `Reader` is just wrapping the `BufferedSource` and it holds no resources itself.

Comment (NightlyNexus) [inline samples/simple-client/src/main/java/com/squareup/okhttp/sample/OkHttpContributors.java]: Oh, yep.  Now, I see that, and it makes obvious sense.  Thanks for the help!

**Comment by swankjesse** [comment]
Thanks!
